libbeat/tests/integration/proc_win.go

```diff
@@ -0,0 +1,120 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build windows
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func stopCmd(p *os.Process) error {
+	return windows.GenerateConsoleCtrlEvent(windows.CTRL_BREAK_EVENT, uint32(p.Pid))
+}
+
+func getSysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		// This disables the child from receiveing CTRL_C events
+		// But isolates other siganls from us, aka the parent
+		CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+	}
+}
+
+// Job is wrapper for windows JobObject
+// https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects
+// This helper guarantees a clean process tree kill on job handler close
+type Job windows.Handle
+
+// CreateJobObject creates JobObject on Windows, global per process
+// Should only be initialized once in main function
+func CreateJobObject() (pj Job, err error) {
+	if pj, err = NewJob(); err != nil {
+		return pj, err
+	}
+
+	return pj, nil
+}
+
+// NewJob creates a instance of the JobObject
+func NewJob() (Job, error) {
```

Comment:
This code is actually already in beats twice: https://github.com/elastic/beats/blob/8e442a88fb9a7adc6931af38ee41c9146a96e5e3/x-pack/libbeat/common/proc/job_windows.go#L37

It's used by osquerybeat and it out of date with the change in agent.

Re-reviewing this I think it's fine to have this job object code in here copied from agent, and you can also Apache license it. Bring it in as a separate PR and put it in libbeat and have the osquerybeat code that needs it call into it: https://github.com/elastic/beats/blob/8e442a88fb9a7adc6931af38ee41c9146a96e5e3/x-pack/osquerybeat/beater/osquerybeat.go#L132